gymnasium_robotics/envs/multiagent_mujoco/mujoco_multi.py

```diff
@@ -0,0 +1,636 @@
+    agent_conf="2x1"
+
+    ### 2-Agent Swimmer
+
+    scenario="Swimmer-v2"
+    agent_conf="2x1"
+
+    ### 2-Agent Walker
+
+    scenario="Walker2d-v2"
+    agent_conf="2x3"
+
+    ### 1-Agent InvertedPendulum (for debugging algorithms)
+    scenario="InvertedPendulum"
+    agent_conf=None
+
+    ### Manyagent Swimmer
+
+    scenario="manyagent_swimmer"
+    agent_conf="10x2"
+
+    scenario="manyagent_swimmer"
+    agent_conf="$Xx$Y" # where $X, $Y any positive integers e,g, "42x6", "10x2", "2x3"
+
+
+    ### Manyagent Ant
+
+    scenario="manyagent_ant"
+    agent_conf="2x3"
+
+    scenario="manyagent_ant"
+    agent_conf="$Xx$Y" # where $X, $Y any positive integers e,g, "42x6", "10x2", "2x3"
+
+    ### Coupled HalfCheetah (NEW!)
+
+    scenario="coupled_half_cheetah"
+    agent_conf="1p1"
+
+
+    # How to create new agent factorizations (example 'Ant-v4', '8x1')
+
+    In this example, we will create an agent factorization not present in MaMuJoCo the '8x1', where each agent controls a single action (first implemented by [safe-MaMuJoCo](https://github.com/chauncygu/Safe-Multi-Agent-Mujoco))
+
+    first we will load the graph of MaMuJoCo
+    ```python
+    >>> from multiagent_mujoco.obsk import get_parts_and_edges
+    >>> unpartioned_nodes, edges, global_nodes = get_parts_and_edges('Ant-v4', None)
+    ```
+    the `unpartioned_nodes` contain the nodes of the MaMuJoCo graph
+    the `edges` well, contain the edges of the graph
+    and the `global_nodes` a set of observations for all agents
+
+    To create our '8x1' partition we will need to partition the `unpartioned_nodes`
+
+    ```python
+    >>> unpartioned_nodes
+    [(hip1, ankle1, hip2, ankle2, hip3, ankle3, hip4, ankle4)]
+    >>> partioned_nodes = [(unpartioned_nodes[0][0],), (unpartioned_nodes[0][1],), (unpartioned_nodes[0][2],), (unpartioned_nodes[0][3],), (unpartioned_nodes[0][4],), (unpartioned_nodes[0][5],), (unpartioned_nodes[0][6],), (unpartioned_nodes[0][7],)]>>> partioned_nodes
+    >>> partioned_nodes
+    [(hip1,), (ankle1,), (hip2,), (ankle2,), (hip3,), (ankle3,), (hip4,), (ankle4,)]
+    ```
+    finally package the partitions and create our environment
+    ```python
+    my_agent_factorization = {"partition": partioned_nodes, "edges": edges, "globals": global_nodes}
+    gym_env = MaMuJoCo('Ant', '8x1', agent_factorization=my_agent_factorization)
+    ```
+    """
+
+    metadata = {
+        "render_modes": ["human", "rgb_array", "depth_array"],
+        "name": "MaMuJoCo",
+        "is_parallelizable": True,
+        # "render_fps": 0,  #depends on underlying Envrioment
+        "has_manual_policy": False,
+    }
+
+    def __init__(
+        self,
+        scenario: str,
+        agent_conf: str,
+        agent_obsk: int = 1,
+        agent_factorization: dict[str, any] = None,
+        local_categories: list[list[str]] = None,
+        global_categories: list[str] = None,
+        render_mode: str = None,
+    ):
+        """
+        Args:
```

Comment:
This comment is close to the google docstring but you need to update the raises section and parameter names parts
https://google.github.io/styleguide/pyguide.html#doc-function-raises

Reply:
Literally scroll down 1 page on the link you sent me -:)

Reply:
Never mind `pydocstyle` says you were right